### PR TITLE
Implement filtering by chat type

### LINKED ISSUE #105: feat: Filtering for chat type
We should add `composer.chatType` which takes a `ChatType | ChatType[]` and filters for the given chat types.

### PR DESCRIPTION
**Pull request by KnorpelSenf**
TODO:

- [x] narrow down context type
- [x] add JSDoc

Closes #105.

### CODE REVIEW

**Comment by KnorpelSenf** [comment]
This is blocked by #100. The narrowed context for chat types should be exposed. Will wait in order to prevent merge conflicts.

**Comment by KnorpelSenf** [comment]
Types should ideally also reflect that channel posts are sent in channels, and messages aren't. There may be a lot more dependencies between these things. This will get a very complex type annotation, but it is probably worth it.

File: src/composer.ts
Comment (KnorpelSenf) [reply src/composer.ts:3]: Yes. The compilation process will take care of it.

**Review by kolay-v — APPROVED** [review]

**Comment by KnorpelSenf** [comment]
Thanks for the review and the suggestion about optional types! I'll merge this as soon as we have a second approval :)

@all-contributors add @kolay-v for review

File: src/composer.ts
Comment (EdJoPaTo) [inline src/composer.ts:439]: I think there arent many usecases for using the middleware directly? Normally it will be a combination of filters like command, callbackQuery and so on. So the docs here should prefer using other commands rather than showing the usage with the middleware?

**Review by EdJoPaTo — COMMENTED** [review]
I like this as it both simplifies some code and provides better typings. Worked in the bots I tried it both without changes and with using the new `chatType` method.

File: src/composer.ts
Comment (KnorpelSenf) [reply src/composer.ts:439]: I am not 100 per cent sure what you mean, are you suggesting to put this example higher up in the JSDoc comment, and skip the part about using `chatType` on its own?

Comment (EdJoPaTo) [reply src/composer.ts:439]: Most people will use it like this: `bot.chatType("private").command("start")`.

I don't see a real use case to use it like this: `bot.chatType("private", ctx => { ... })`. (Maybe except [stuff like this](https://github.com/HAWHHCalendarBot/TelegramBot/blob/3f9753c567694751006ef30ffc2101e3951d4388/source/parts/easter-eggs.ts#L15-L19))

I think the documentation should show prominently ("higher up") what most users will want to use.

Comment (KnorpelSenf) [reply src/composer.ts:439]: Good point, I'll fix it

File: src/composer.ts
Comment (EdJoPaTo) [inline src/composer.ts:429]:
```suggestion
     * example, this is how can you respond to `/start` commands only from
     * private chats:
```

**Review by EdJoPaTo — APPROVED** [review]

**Comment by KnorpelSenf** [comment]
Thanks for the reviews!

**Pull request merged**
